plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -502,44 +557,38 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
          trace_log->add_transaction(p, t);
    }
 
-   void store(const block_state_ptr& block_state) {
+   void store(const block_state_ptr& block_state, const ::std::optional<::fc::zipkin_span>& ship_accept_span) {
       try {
-         if (trace_log)
+         if (trace_log) {
+            auto trace_log_span = fc_create_span(ship_accept_span, "store_trace_log");
             trace_log->store(chain_plug->chain().db(), block_state);
-         if (chain_state_log)
+         }
+         if (chain_state_log) {
+            auto delta_log_span = fc_create_span(ship_accept_span, "store_delta_log");
             chain_state_log->store(chain_plug->chain().kv_db(), block_state);
+         }
          return;
       }
       FC_LOG_AND_DROP()
 
-      // Both app().quit() and exception throwing are required. Without app().quit(),
-      // the exception would be caught and drop before reaching main(). The exception is
-      // to ensure the block won't be committed.
-      appbase::app().quit();
-      EOS_THROW(
-          chain::state_history_write_exception,
-          "State history encountered an Error which it cannot recover from.  Please resolve the error and relaunch "
+      elog("State history encountered an Error which it cannot recover from.  Please resolve the error and relaunch "
           "the process");
+      exit(1);
```

Comment:
This will cause a dirty db. Seems like we need a way to shutdown without a dirty db.